homepage/tests.py

```diff
@@ -103,3 +106,145 @@ def test_create_multiple_courses():
     # make sure that multiple courses are saved accurately
     courses = [Course.objects.get(pk=arg[0]).get_details() for arg in details]
     assert courses == details
+
+
+# ---------Professor tests----------#
+
+@pytest.fixture
+def professors_list():
+    return list(Professor.objects.values_list(
+        'name'))
+
+
+@pytest.fixture
+def valid_name():
+    return 'Severus Snape'
+
+
+@pytest.fixture
+def invalid_name():
+    return 'Pablo Diego Jose Francisco de Paula Juan Nepomuceno Maria de \
+    los Remedios Cipriano de la Santisima Trinidad Ruiz y Picasso Snape'
+
+
+# check the data stored in database from 0004_professor_test_data.py file
+@pytest.mark.django_db
+def test_model_professors_list(professors_list):
+
+    assert professors_list == [
+        ('Septima Vector',),
+        ('Sybill Patricia Trelawney',),
+        ('Bathsheda Babbling',),
+    ]
+
+
+@pytest.mark.django_db
+def test_add_valid_professor(valid_name):
+    professor = Professor(name=valid_name)
+    professor.save()
+
+    assert Professor.objects.filter(pk=professor.id).exists()
+
+
+# name is too long (>100)
+@pytest.mark.django_db
+def test_create_invalid_professor(invalid_name):
+    invalid = False
+    try:
+        Professor(invalid_name).clean_fields()
+    except ValidationError:
+        invalid = True
+
+    assert invalid
+
+
+@pytest.mark.django_db
+def test_professor_str(valid_name):
+    professor = Professor(name=valid_name)
+
+    assert str(professor) == professor.name
+
+
+# -----Professor to course tests-----#
+
+@pytest.fixture
+def professor_to_course_list():
+    return list(Professor_to_Course.objects.values_list(
+        'professor_id', 'course_id'))
+
+
+# check the data stored in database from 0004_professor_test_data.py file
+@pytest.mark.django_db
+def test_professor_to_course_list(professor_to_course_list):
+
+    assert professor_to_course_list == [
+        (1, 10221),
+        (1, 12357),
+        (2, 10231),
+        (3, 10111)
+    ]
+
+
+@pytest.mark.django_db
+def test_add_valid_pro_to_course():
+    relation = Professor_to_Course(
+        professor_id=Professor.objects.get(pk=2),
+        course_id=Course.objects.get(pk=10231))
+    relation.save()
+

```

Comment:
same as adding a valid professor - consider adding a check that it doesn't raise an exception?

Reply:
Same ^^